File: Patrones Creacionales/Abstract Factory/Ejemplo.py
class Button:
    def render(self): pass

class WindowsButton(Button):
    def render(self):
        return "Boton Windows"

class MacButton(Button):
    def render(self):
        return "Boton Mac"

class GUIFactory:
    def create_button(self): pass

class WindowsFactory(GUIFactory):
    def create_button(self):
        return WindowsButton()

class MacFactory(GUIFactory):
    def create_button(self):
        return MacButton()

factory = WindowsFactory()
print(factory.create_button().render())

File: Patrones Creacionales/Abstract Factory/Ejemplo.js
class Button {
  render() {}
}

class WindowsButton extends Button {
  render() { 
    return "Boton Windows"; 
  }
}

class MacButton extends Button {
  render() { 
    return "Boton Mac"; 
  }
}

class GUIFactory {
  createButton() {}
}

class WindowsFactory extends GUIFactory {
  createButton() { 
    return new WindowsButton(); 
  }
}

class MacFactory extends GUIFactory {
  createButton() { 
    return new MacButton(); 
  }
}

const factory = new WindowsFactory();
console.log(factory.createButton().render());
